core/server/update-check.js

```diff
@@ -161,16 +163,16 @@ function updateCheckResponse(response) {
         api.settings.edit(
             {settings: [{key: 'nextUpdateCheck', value: response.next_check}]},
             internal
-        ).catch(errors.rejectError),
+        ).catch(errors.rejectError).reflect(),
         api.settings.edit(
             {settings: [{key: 'displayUpdateNotification', value: response.version}]},
             internal
-        ).catch(errors.rejectError)
+        ).catch(errors.rejectError).reflect()
     );
 
-    return Promise.settle(ops).then(function then(descriptors) {
+    return Promise.map(ops).then(function then(descriptors) {
```

Comment:
`Promise.map` needs to be called like `Promise.map(Array, fn)`. Omitting its callback function is the source of the error.

That said, I see what you're trying to do here but even if you fix the way `Promise.map` is being used, if any of the promises in `ops` rejects, `Promise.map` will return a rejected promise with the `reason` of the first rejected promise it encounters as its `reason`. So `updateCheckError` will still end up with only a single promise as its argument.

Some possible solutions:

* Just use `return Promise.all(ops);` and fail fast if any of the promises reject. This would also mean removing the `.catch(errors.rejectError).reflect()` from each of the `api.settings.edit()` calls. If both settings calls fail you'd obviously swallow one of the error messages but given the intent of this flow I'm not sure it's a huge concern.
* Or, also using `Promise.all` with `.reflect()` (you could still remove the `.catch(errors.rejectError)`), iterate through the return values (`descriptors` in the above code), check for rejections and build a composite `Error` object that contains all of the error messages and then return a rejection with that (or not, if there weren't any errors).